src/library_dylink.js

```diff
@@ -694,32 +688,24 @@ var LibraryDylink = {
       return;
     }
 
-    // if we can load dynamic libraries synchronously, do so, otherwise, preload
-    if (!readBinary) {
-      // we can't read binary data synchronously, so preload
-      addRunDependency('preloadDylibs');
-      dynamicLibraries.reduce(function(chain, lib) {
-        return chain.then(function() {
-          return loadDynamicLibrary(lib, {loadAsync: true, global: true, nodelete: true, allowUndefined: true});
-        });
-      }, Promise.resolve()).then(function() {
-        // we got them all, wonderful
-        removeRunDependency('preloadDylibs');
-        reportUndefinedSymbols();
+    // Load binaries asynchronously
+    addRunDependency('preloadDylibs');
+    dynamicLibraries.reduce(function(chain, lib) {
+      return chain.then(function() {
+        return loadDynamicLibrary(lib, {loadAsync: true, global: true, nodelete: true, allowUndefined: true});
       });
-      return;
-    }
-
-    dynamicLibraries.forEach(function(lib) {
```

Comment:
I don't see this logic anywhere in the new version, is that intentional?

Reply:
Yes, now that readAsync, is ubiquitous we can just have single code path there that works everywhere.  No need for the extra code path above.